x-pack/plugins/cases/public/components/create/form_context.tsx

```diff
@@ -89,6 +94,33 @@ export const FormContext: React.FC<Props> = ({
     return formData;
   };
 
+  const transformCustomFieldsData = useCallback(
+    (customFields: { [x: `customFields.${string}`]: string | boolean }) => {
+      const transformedCustomFields = [];
+
+      if (!customFields || !customFieldsConfiguration.length || isLoadingCaseConfiguration) {
+        return [];
+      }
+
+      for (const [key, value] of Object.entries(customFields)) {
+        const configCustomField = customFieldsConfiguration.find((item) => item.key === key);
+
+        if (configCustomField) {
+          transformedCustomFields.push({
+            key: configCustomField.key,
+            type: configCustomField.type,
+            field: {
+              value: [value],
```

Comment:
The value should be `const value = isEmpty(value) ? null : [value]`.

Reply:
Had to add` ts-ignore` [8e78b12](https://github.com/elastic/kibana/pull/166969/commits/8e78b121501bef04ce2c2789936e0a3cc912b46d) as typescript complains:
` Type 'string | boolean' is not assignable to type 'string'.
        Type 'boolean' is not assignable to type 'string'`

Reply:
removed ts-ignore and used casting